app/controllers/concerns/foreman/controller/ssh_keys_common.rb

```diff
@@ -0,0 +1,25 @@
+module Foreman::Controller::SshKeysCommon
+  extend ActiveSupport::Concern
+
+  included do
+    before_action :find_user
+  end
+
+  private
+
+  def resource_scope(*args, &block)
+    super.where(:user => @user)
+  end
+
+  def find_user
+    if editing_self?
+      @user = User.current
+    else
+      @user = User.find(params[:user_id])
+    end
+  end
+
+  def editing_self?
```

Comment:
This is duplicating logic already existing in the UsersMixin, extract to a shared concern or see if you can simply reuse the UsersMixin in both SSHKeys and User controllers